test/test_images/latencymako/main.go

```diff
@@ -0,0 +1,471 @@
+	}
+
+	targeter := common.NewCloudEventsTargeter(sinkURL, msgSize, defaultEventType, defaultEventSource, "binary").VegetaTargeter()
+
+	pacers := make([]vegeta.Pacer, len(pacerSpecs))
+	durations := make([]time.Duration, len(pacerSpecs))
+	var totalBenchmarkDuration time.Duration = 0
+
+	for i, ps := range pacerSpecs {
+		pacers[i] = vegeta.ConstantPacer{ps.rps, time.Second}
+		durations[i] = ps.duration
+		printf("%d° pace: %d rps for %v seconds", i+1, ps.rps, ps.duration)
+		totalBenchmarkDuration = totalBenchmarkDuration + ps.duration
+	}
+
+	printf("Total benchmark duration: %v", totalBenchmarkDuration.Seconds())
+
+	combinedPacer, err := pkgpacers.NewCombined(pacers, durations)
+
+	if err != nil {
+		fatalf("Failed to setup combined pacer: %v", err)
+	}
+
+	client := http.Client{Transport: requestInterceptor{before: func(request *http.Request) {
+		id, _ := strconv.ParseUint(request.Header["Ce-Id"][0], 10, 64)
+		sentCh <- sentState{eventId: id, at: time.Now()}
+	}, after: func(request *http.Request, response *http.Response, e error) {
+		id, _ := strconv.ParseUint(request.Header["Ce-Id"][0], 10, 64)
+		if e != nil || response.StatusCode < 200 || response.StatusCode > 300 {
+			deliveredCh <- deliveredState{eventId: id, failed: true}
+		} else {
+			deliveredCh <- deliveredState{eventId: id, at: time.Now()}
+		}
+	}}}
+
+	printf("Starting benchmark")
+
+	vegetaResults := vegeta.NewAttacker(
+		vegeta.Client(&client),
+		vegeta.Workers(workers),
+		vegeta.MaxWorkers(workers),
+	).Attack(targeter, combinedPacer, totalBenchmarkDuration, defaultEventType+"-attack")
+
+	go processVegetaResult(vegetaResults)
+
+	// count errors
+	var publishErrorCount int
+	var deliverErrorCount int
+	for eventState := range resultCh {
+		switch eventState {
+		case dropped:
+			deliverErrorCount++
+		case undelivered:
+			publishErrorCount++
+		}
+	}
+
+	printf("Publish error count: %v", publishErrorCount)
+	printf("Deliver error count: %v", deliverErrorCount)
+
+	printf("--- END BENCHMARK ---")
+
+	// publish error counts as aggregate metrics
+	q.AddRunAggregate("pe", float64(publishErrorCount))
+	q.AddRunAggregate("de", float64(deliverErrorCount))
+
+	out, err := q.Store()
+	if err != nil {
+		fatalf("q.Store error: %v: %v", out, err)
+	}
+}
+
+func parsePaceSpec() ([]paceSpec, error) {
+	paceSpecArray := strings.Split(paceFlag, ",")
+	pacerSpecs := make([]paceSpec, 0)
+
+	for _, p := range paceSpecArray {
+		ps := strings.Split(p, ":")
+		rps, err := strconv.Atoi(ps[0])
+		if err != nil {
+			return nil, fmt.Errorf("Error while parsing pace spec %v: %v", ps, err)
+		}
+		duration := defaultDuration
+
+		if len(ps) == 2 {
+			durationSec, err := strconv.Atoi(ps[1])
+			if err != nil {
+				return nil, fmt.Errorf("Error while parsing pace spec %v: %v", ps, err)
+			}
+			duration = time.Second * time.Duration(durationSec)
+		}
+
+		pacerSpecs = append(pacerSpecs, paceSpec{rps, duration})
+	}
+
+	return pacerSpecs, nil
+}
+
+func warmup() {
```

Comment:
I'd pass the warmup duration to that function instead of relying on the global name, just to make the signature self-explanatory.

Reply:
To be honest I think it's already self explanatory as is, since function name "warmup" already underlines what the function is going to do

Reply:
It says _what it does_, not what the function depends on. Considering that `warmupSeconds` is used only by the `warmup()` function, I think it would be a good coding practice to pass the parameter instead of making it global. This is of course only a suggestion, feel free to ignore.